### DISCUSSION TITLE
How fetchCallback works?

### DISCUSSION THREAD

**Discussion by Divinely5782** [mskocik/svelecte#262] [Q&A] [answered]
Hi! Thanks for this plugin! I am trying to fetch cities from remote api, but this api returns me an array of objects,like:
[
  {
    "id": "5368381",
    "name": "Los Angeles",
    "latitude": 34.19801,
    "population": "9818605",
    "timezone": "America/Los_Angeles",
the rest....

fetchCallback function is:
const convertCities = (response) => {
        return response.map(item => item.name);
  };
  
  <Svelecte fetch="myapi.com/[query]" fetchCallback={convertCities} />
  
 I just want to extract Name and timezone for future usage.
 Many thanks for help. Sorry for possible newbie misunderstanding.

**Comment by mskocik** [comment, marked as answer]
Hi, `fetchCallback` must return array of _objects_ where every object must have property representing value (let's call it `value`) and text (this is `label`), which is being shown to users. 
Svelecte automatically tries to pick following properties if they exist in option objects:

- for `value` one of these properties is tried (in this order): `['id', 'value', 'ID']`
- for `label` this list is used: `['name', 'title', 'label']`

So when we take your example option object, you don't need to do anything, because Svelecte should automatically use `id` property for `value` and `name` for `label`. If you would like to specify some other properties, you can use `valueField` and `labelField` properties to define which properties should be used.
![image](https://github.com/user-attachments/assets/4318ea2c-72d3-4973-bacb-0f920e3dcb8e)


In general `fetchCallback` should be used, when you need to convert fetched data somehow before using in Svelecte. For example, you need to "compute" `label` field from other properties, etc. Is that clear now?

**Reply by Divinely5782** [reply]
Aaahh) Now it's perfectly clear and working as expected. Appreciate your work and wish you success! 

One more little question, after selecting a city, cursor is still inside input field, suggesting to enter next value, how do i limit it to allow only one select? Didnt find an answer in options. Thanks again!
